.github/workflows/ci.yml

```diff
@@ -1,68 +1,182 @@
 name: Build
 
 on:
-  push:
-    branches: [ 'main*' ]
-    paths-ignore:
-    - '**.md'
   pull_request:
     branches: [ 'main*' ]
-    paths-ignore:
-    - '**.md'
 
 jobs:
-  build-test-stable:
-    strategy:
-      fail-fast: false  # ensures the entire test matrix is run, even if one permutation fails
-      matrix:
-        os: [ windows-latest, ubuntu-latest ]
-        version: [ net462, net6.0, net7.0, net8.0 ]
-        exclude:
-        - os: ubuntu-latest
-          version: net462
+  lint-misspell-sanitycheck:
+    uses: ./.github/workflows/sanitycheck.yml
 
-    runs-on: ${{ matrix.os }}
+  detect-changes:
+    runs-on: windows-latest
+    outputs:
+      changes: ${{ steps.changes.outputs.changes }}
     steps:
-    - uses: actions/checkout@v4
+    - uses: AurorNZ/paths-filter@v3
+      id: changes
       with:
-        fetch-depth: 0 # fetching all
+        filters: |
+          md: ['**.md']
+          build: ['build/**', '.github/**/*.yml', '**/*.targets', '**/*.props']
+          shared: ['src/Shared/**']
+          code: ['**.cs', '**.csproj', '.editorconfig']
+          packaged-code: ['src/**', '!**/*.md']
+          api-code: ['*/OpenTelemetry.Api*/**', '!**/*.md']
+          api-packages: ['src/OpenTelemetry.Api*/**', '!**/*.md']
+          instrumentation: ['*/OpenTelemetry.Instrumentation*/**', 'test/TestApp.AspNetCore/**', '!**/*.md']
```

Comment:
given this repo only has well defined components this is okay. The contrib repo needs some changes to the wording in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#contributing-a-new-project I believe, so that new components get CI covered. Not a blocker for this PR, just something to be addressed in contrib repo.

Reply:
Any new component added in the contrib repo would still get CI covered. It's just that if the new component does not a component specific `.proj` [file](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/build/Projects), any changes to the component would trigger CI for every project in the solution that does not have a dedicated `proj` file. That's because the [filtering logic](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/ci.yml#L38-L65) would determine that the [CI needs to run for every project that does not have a dedicated proj file](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/ci.yml#L233-L309).

Reply:
We probably should update the CONTRIBUTING.md to mention that new contributios could have a more efficient CI check by introducing a dedicated `proj` file for their component.